ArduPlane/tiltrotor.cpp

```diff
@@ -543,13 +543,21 @@ void Tiltrotor::vectoring(void)
 
     const bool no_yaw = tilt_over_max_angle();
     if (no_yaw) {
+        Vector2f scaler;
         // fixed wing  We need to apply inverse scaling with throttle, and remove the surface speed scaling as
         // we don't want tilt impacted by airspeed
-        const float scaler = plane.control_mode == &plane.mode_manual?1:(quadplane.FW_vector_throttle_scaling() / plane.get_speed_scaler());
-        const float gain = fixed_gain * fixed_tilt_limit * scaler;
-        const float right = gain * SRV_Channels::get_output_scaled(SRV_Channel::k_elevon_right) * (1/4500.0);
-        const float left  = gain * SRV_Channels::get_output_scaled(SRV_Channel::k_elevon_left) * (1/4500.0);
-        const float mid  = gain * SRV_Channels::get_output_scaled(SRV_Channel::k_elevator) * (1/4500.0);
+        if (plane.control_mode == &plane.mode_manual) {
+            scaler.x = scaler.y = 1.0f;
+        } else {
+            const Vector3f speed_scaler_rpy = plane.get_rpy_speed_scaler();
+            scaler.x = quadplane.FW_vector_throttle_scaling() / speed_scaler_rpy.x;
+            scaler.y = quadplane.FW_vector_throttle_scaling() / speed_scaler_rpy.y;
+        }
+        const Vector2f gain = scaler * (fixed_gain * fixed_tilt_limit);
+        const float elevon_gain = (gain.x + gain.y) * 0.5f;
```

Comment:
This is a little odd, it assumes a fixed mixing gain, which might not be the case.

Reply:
Thinking about this some more, the method probably isn't going to work for tilt rotors so will remove this.

Reply:
The user should not be using this method of speed scaling for tilt rotors because the thrust line is not aligned with the X axis. I will add code to prevent the slipstream compensation running for tilt rotor platforms and also update documentation. We can then use just the x component.

Reply:
The simplest way of handling this would be to not doe forward flight slipstream compensation when flight_stage == AP_FixedWing::FlightStage::VTOL. Tailsitter already has its own compensation for VTOL flight.